### PR TITLE
rearrange error setter close #22

### LINKED ISSUE #22: Stripe Card Error thrown when invalid card entered "The default action was triggered "Assertion Failed: The default action was triggered on the component... but the action name ([object Object]) was not a string."
In template:

```
{{stripe-card}}
```

Enter an intentionally wrong card number (such as "8768 7658 7686 8766"), and the console shows an exception: 

> Assertion Failed: The default action was triggered on the component [component id], but the action name ([object Object]) was not a string.

image

image

### PR DESCRIPTION
**Pull request by snewcomer**
@joshsmith 

close #22

### CODE REVIEW

**Comment by joshsmith** [comment]
Is there anything we can do in order to test this? At the integration level perhaps? Even if it's just a regression smoke test.

**Comment by snewcomer** [comment]
Maybe.  It's quite tough to test these b/c the need to load the stripe script and elements.

**Comment by joshsmith** [comment]
Yeah, I mean the hardest thing is testing the internals of it because it's in an iframe. But the `error` being set should be possible to test without actually testing the elements.

**Comment by joshsmith** [comment]
Maybe in the interest of getting the fix merged we can open a follow-up issue to test it?

We do have a `StripeMock` that we can use that doesn't require us to specifically have the script loading.

**Comment by joshsmith** [comment]
So testing this now myself, this just prevents the error the first time, but does not prevent it from recurring again (since `error` is now set to the object on the component, rather than an action).

**Comment by snewcomer** [comment]
Ah then we can just change the name?  `errorObj`?

**Comment by joshsmith** [comment]
I think we can revert your change and change `error` to `stripeError`. This will need changed in the docs, `tests/dummy/app/templates/application.hbs`, the component js, and each component's template.

Should have ~8 file changes by the end of it.

**Comment by snewcomer** [comment]
Yeah I'll create another issue to figure out testing tomorrow.

**Comment by joshsmith** [comment]
The README needs changed, too.

File: addon/components/stripe-element.js [hunk 1]
```diff
@@ -6,7 +6,7 @@ export default Component.extend({
   classNames: ['ember-stripe-element'],
 
   autofocus: false,
-  error: null,
+  stripeError: null,
```
Comment (joshsmith) [inline addon/components/stripe-element.js]: I would reorder this by alpha now.

File: addon/components/stripe-element.js [hunk 2]
```diff
@@ -68,16 +68,16 @@ export default Component.extend({
     stripeElement.on('blur',    (event) => this.sendAction('blur', stripeElement, event));
     stripeElement.on('focus',   (event) => this.sendAction('focus', stripeElement, event));
     stripeElement.on('change',  (...args) => {
-      let { error, complete } = args[0];
+      let { error: stripeError, complete } = args[0];
```
Comment (joshsmith) [inline addon/components/stripe-element.js]: I think we should array destructure here while we have the chance.

```
let [{ complete, error: stripeError }] = args;
```

**Comment by joshsmith** [comment]
Looks good! Rebase and squash?

**Comment by joshsmith** [comment]
Actually, one sec. I think a change I asked about was missed.

File: addon/components/stripe-element.js [hunk 3]
```diff
@@ -68,15 +68,16 @@ export default Component.extend({
     stripeElement.on('blur',    (event) => this.sendAction('blur', stripeElement, event));
     stripeElement.on('focus',   (event) => this.sendAction('focus', stripeElement, event));
     stripeElement.on('change',  (...args) => {
-      let { error, complete } = args[0];
-      set(this, 'error', error);
+      let { error: stripeError, complete } = args[0];
```
Comment (joshsmith) [inline addon/components/stripe-element.js]: Reposting:

I think we should array destructure here while we have the chance.

```
let [{ complete, error: stripeError }] = args;
```

**Comment by snewcomer** [comment]
@joshsmith I think I fixed everything!  And added a simple test to yield out the stripeError.  I can work on more tests tomorrow.

File: tests/integration/components/stripe-card-test.js [hunk 4]
```diff
@@ -28,3 +28,14 @@ test('it renders', function(assert) {
...
+
+test('yields out error message', function(assert) {
+  this.stripeError = { message: 'oops' };
+  this.render(hbs`
+    {{#stripe-card stripeError=stripeError as |stripeElement stripeError|}}
+      {{stripeError.message}}
+    {{/stripe-card}}
+  `);
+
+  assert.equal(this.$().text().trim(), 'oops');
```
Comment (joshsmith) [inline tests/integration/components/stripe-card-test.js]: Great! Glad this made it in so we're sure we fixed it.

**Pull request merged**